src/app/modules/submission/ParsedResponsesObject.class.ts

```diff
@@ -145,18 +141,7 @@ export default class ParsedResponsesObject {
 
       const processingResponse: ProcessedFieldResponse = {
         ...response,
-        isVisible:
-          // Set isVisible as true for Encrypt mode if there is a response for mobile and email field
-          // Because we cannot tell if the field is unhidden by logic
-          // This prevents downstream validateField from incorrectly preventing
-          // encrypt mode submissions with responses on unhidden fields
-          // TODO(#780): Remove this once submission service is separated into
-          // Email and Encrypted services
-          form.responseMode === FormResponseMode.Encrypt
-            ? 'answer' in response &&
-              typeof response.answer === 'string' &&
-              response.answer.trim() !== ''
-            : visibleFieldIds.value.has(responseId),
+        isVisible: visibleFieldIds.value.has(responseId),
         question: formField.getQuestion(),
```

Comment:
sorry the preceeding comments (before the diff) are a bit hard to decipher :sweat. would you mind just giving a quick 1-liner on what this change does (if any)

Reply:
Okay so the comment was added back when email and encrypted services are not separate yet, I think. So the TODO was there to remove the conditional once it has been separated, which it already has been if we track the issue #780 tagged. So this is a dangling conditional that should have been removed a while back.

Since encrypt mode (ie old submission endpoint) does not use `ParsedResponsesObject`, we can simply remove the conditional to set the `isVisible` field to what it was being set for email mode, which is `visibleFieldIds.value.has(responseId)`.

Reply:
got it, thanks @LinHuiqing !